util/Pending.h

```diff
@@ -87,7 +87,7 @@ namespace Pending {
     boost::optional<T> WaitForPending(boost::optional<Pending<T>>& pending, bool do_not_care_about_result = false) {
         if (!pending)
             return boost::none;
-        std::lock_guard<std::mutex> lock(pending->m_mutex);
+        std::scoped_lock lock(pending->m_mutex);
```

Comment:
this scoped_lock is also not strictly necessary. WaitForPending and SwapPending are the only places where a pending->m_mutex gets locked and there is no code path where one calls the other. OTH the only downside of the scoped_lock is performance (if it works correctly on all target platforms).